xarray/core/computation.py

```diff
@@ -1299,3 +1299,49 @@ def where(cond, x, y):
         dataset_join="exact",
         dask="allowed",
     )
+
+
+def polyval(coord, coeffs, degree_dim="degree"):
+    """Evaluate a polynomial at specific values
+
+    Parameters
+    ----------
+    coord : DataArray or array-like
+        The 1D coordinate along which to evaluate the polynomial.
+    coeffs : DataArray or array-like
+        Coefficients of the polynomials.
+    degree_dim : str or int, optional
+        Name of dimension or axis number along which the degree increases in `coeffs`.
```

Comment:
Is there a reason why we are allowing the passing of axis number here instead of just dimension string? We don't do this anywhere else in xarray's API afaik

Reply:
I wanted the user to be able to pass a coefficient `ndarray` instead of a `DataArray`, thus needing an axis number. But if this too different form the rest of the api, I can remove it!

Reply:
I think that if no other information apart from the values is required, then it's okay to use duck-typing to accept a general array-like object (so `weights` is probably okay to be general). But for `coord` and `coeffs` extra information is required which would normally be part of the `xarray.DataArray`, so best to limit the input to xarray objects for those arguments. (It will also simplify your code and tests)